src/main/webapp/app/exercises/shared/exercise-scores/exercise-scores-export-button.component.ts

```diff
@@ -0,0 +1,84 @@
+
+@Component({
+    selector: 'jhi-exercise-scores-export-button',
+    template: `
+        <button class="btn btn-info btn-sm mr-1" (click)="exportResults()">
+            <fa-icon [icon]="'download'"></fa-icon>
+            <span class="d-none d-md-inline" jhiTranslate="artemisApp.exercise.exportResults">Export Results</span>
+        </button>
+    `,
+})
+@Injectable({ providedIn: 'root' })
+export class ExerciseScoresExportButtonComponent {
+    @Input() exercises: Exercise[] = []; // Used to export multiple scores together
+    @Input() exercise: Exercise | ProgrammingExercise;
+
+    constructor(private resultService: ResultService, private jhiAlertService: JhiAlertService) {}
+
+    /**
+     * Exports the exercise results as a csv file
+     */
+    exportResults() {
+        if (this.exercises.length === 0 && this.exercise !== undefined) {
+            this.exercises = this.exercises.concat(this.exercise);
+        }
+        this.exercises.forEach((exercise) => {
+            this.exercise = exercise;
+            this.resultService.getResults(exercise).subscribe((data) => {
+                const rows: string[] = [];
+                const results = data.body || [];
+                if (results.length === 0) {
+                    this.jhiAlertService.warning(`artemisApp.exercise.exportResultsEmptyError`, { exercise: exercise.title });
+                    window.scroll(0, 0);
+                    return;
+                }
+                results.forEach((result, index) => {
+                    const studentParticipation = result.participation! as StudentParticipation;
+                    const { participantName, participantIdentifier } = studentParticipation;
+                    const score = round(result.score);
+
+                    if (index === 0) {
+                        rows.push(this.getHeadersRow(studentParticipation));
+                    }
+                    const optionalStudentsColumnValue = studentParticipation.team ? `,"${studentParticipation.team?.students?.map((s) => s.name).join(', ')}"` : '';
+                    if (this.exercise.type !== ExerciseType.PROGRAMMING) {
+                        rows.push(`${participantName},${participantIdentifier},${score}${optionalStudentsColumnValue}`);
+                    } else {
+                        const repoLink = (studentParticipation as ProgrammingExerciseStudentParticipation).repositoryUrl;
+                        rows.push(`${participantName},${participantIdentifier},${score},${repoLink}${optionalStudentsColumnValue}`);
+                    }
+                });
+                const csvContent = rows.join('\n');
+                const encodedUri = encodeURI(csvContent);
+                const link = document.createElement('a');
+                link.setAttribute('href', encodedUri);
+                link.setAttribute('download', `${exercise.shortName}-results-scores.csv`);
+                document.body.appendChild(link); // Required for FF
```

Comment:
Do we have to remove the link element from the body again after clicking? (This code seems to be just existing code which was moved, so not too important to address this imho.)

Reply:
Removing it now after click :+1: